spec/kissable/ab_spec.rb

```diff
@@ -63,67 +64,84 @@
-    end
+      context "when cookie exists" do
+        before :each do
+          ab_test.stub(:cookies).and_return('abid' => 1)
+        end
 
-    context "when cookie doesn't exist" do
-      it "sets a cookie" do
-        expect { group }.to change { ab_test.cookies }.from({})
+        it "doesn't change the cookie" do
+          expect { group }.to_not change { ab_test.cookies }
+        end
       end
 
-      describe "the cookie" do
-        let(:cookie) { ab_test.cookies['abid'] }
+      context "when cookie doesn't exist" do
+        it "sets a cookie" do
+          expect { group }.to change { ab_test.cookies }.from({})
+        end
+
+        describe "the cookie" do
+          let(:cookie) { ab_test.cookies['abid'] }
+
+          context "when the domain has been configured" do
+            let(:domain) { 'someaweseomedomain.com' }
 
-        context "when the domain has been configured" do
-          let(:domain) { 'someaweseomedomain.com' }
+            it "has the domain key set to the correct value" do
+              Kissable.configure do |config|
+                config.domain = domain
+              end
 
-          it "has the domain key set to the correct value" do
-            Kissable.configure do |config|
-              config.domain = domain
+              group
+              expect(cookie).to include(:domain => domain)
             end
+          end
+
+          context "when the domain hasn't been configured" do
+            it "doesn't include a domain key" do
+              Kissable.configure do |config|
+                config.domain = nil
+              end
 
+              group
+              expect(cookie).to_not include(:domain)
+            end
+          end
+
+          it "expires" do
             group
-            expect(cookie).to include(:domain => domain)
+            expect(cookie).to include(:expires)
           end
-        end
 
-        context "when the domain hasn't been configured" do
-          it "doesn't include a domain key" do
-            Kissable.configure do |config|
-              config.domain = nil
-            end
+          it "has a path" do
+            group
+            expect(cookie).to include(:path => "/")
+          end
 
+          it "contains a value" do
             group
-            expect(cookie).to_not include(:domain)
+            expect(cookie).to include(:value)
           end
         end
+      end
+    end
 
-        it "expires" do
-          group
-          expect(cookie).to include(:expires)
-        end
+    context "when using login to group" do
+      let(:group) { ab_test.group(login) }
 
-        it "has a path" do
-          group
-          expect(cookie).to include(:path => "/")
-        end
+      it "returns Original or Variant" do
+        expect(group).to match(/Original|Variant/)
+      end
 
-        it "contains a value" do
-          group
-          expect(cookie).to include(:value)
+      it "returns the same group for an email" do
```

Comment:
When writing Rspec tests, you should read http://betterspecs.org/

Keep in mind, when this test is ran it will say the following:
`Kissable::AB#group when using login to group returns the same group for an email`

I think that this would make more sense:
`Kissable::AB#group when using login to group always returns the same group`